tests/template/test_new_schema_template.py

```diff
@@ -0,0 +1,121 @@
+                    "description": "The numerical value in Timecourse unit",
+                    "pattern": "^[0-9]+\\.?[0-9]*-?[0-9]*\\.?[0-9]*$",
+                    "type": "string",
+                    "example": "2; 5.5-10.5",
+                    "user_friendly": "Timecourse value",
+                    "guidelines": "Enter either a single value or a range of values. Indicate a range using a hyphen."
+                },
+                "unit": {
+                    "description": "The unit in which the Timecourse value is expressed.",
+                    "type": "object",
+                    "user_friendly": "Timecourse unit"
+                },
+            }
+        }
+
+        schema_template = NewSchemaTemplate(json_schema_docs=[sample_metadata_schema_json])
+
+        expected_schema_version = "1.0.0"
+        expected_schema_metadata_properties = {
+            "timecourse": NewSchemaParser(sample_metadata_schema_json).schema_dictionary}
+        expected_schema_labels = {"timecourse.value": ["timecourse.value"], "timecourse.unit": ["timecourse.unit"],
+                                  "timecourse value": ["timecourse.value"], "timecourse unit": ["timecourse.unit"]}
+        expected_schema_tabs = [{"timecourse": {"display_name": "Timecourse"},
+                                 "columns": ["timecourse.value", "timecourse.unit"]}]
+
+        self.assertEqual(schema_template.template_version, expected_schema_version)
+        self.assertEqual(schema_template.meta_data_properties, expected_schema_metadata_properties)
+        self.assertEqual(schema_template.labels, expected_schema_labels)
+        self.assertEqual(schema_template.tabs, expected_schema_tabs)
+
+    def test__lookup_property_in_schema__success(self):
+        sample_metadata_schema_json = {
+            "$schema": "http://json-schema.org/draft-07/schema#",
+            "$id": "https://schema.humancellatlas.org/module/biomaterial/2.0.2/timecourse",
+            "description": "Information relating to a timecourse.",
+            "type": "object",
+            "properties": {
+                "value": {
+                    "description": "The numerical value in Timecourse unit",
+                    "pattern": "^[0-9]+\\.?[0-9]*-?[0-9]*\\.?[0-9]*$",
+                    "type": "string",
+                    "example": "2; 5.5-10.5",
+                    "user_friendly": "Timecourse value",
+                    "guidelines": "Enter either a single value or a range of values. Indicate a range using a hyphen."
+                },
+                "unit": {
+                    "description": "The unit in which the Timecourse value is expressed.",
+                    "type": "object",
+                    "user_friendly": "Timecourse unit"
+                },
+            }
+        }
+
+        schema_template = NewSchemaTemplate(json_schema_docs=[sample_metadata_schema_json])
+
+        expected_property_value = SimplePropertyDescriptor({
```

Comment:
I'm wondering why do you have to instantiate a new SimplePropertyDescriptor object here? Are you planning to have the `lookup_property_attributes_in_metadata` method return SimplePropertyDescriptor object instead of dictionary?

Reply:
Nope. This is just an easy way to verify that the property is formatted correctly. In the assert statement itself, the dictionary representation of the object is compared.